src/main/java/org/candlepin/resource/OwnerProductResource.java

```diff
@@ -154,7 +154,7 @@ public ProductDTO createProductByOwner(String ownerKey, ProductDTO dto) {
         }
 
         if (dto.getName() == null || dto.getName().matches("^\\s*$")) {
-            throw new BadRequestException(i18n.tr("product has null or invalid name"));
+            throw new BadRequestException(i18n.tr("product has a null or invalid name"));
```

Comment:
Updated this to match the first BadRequest message on line 153 so we can assert against both messages in the same spec test.